# Motif-Dock
Motif-Dock is a protein-protein interaction (PPI) search pipeline for proteins containing **specific substructures (3D)** or **sequence motifs (2D)**. Motif-Dock can be used to search for potential PPI partners for specific receptors or protein scaffolds fusing specific motifs such as antigenic epitopes.  


Introduction
----

A number of proteins can bind to the same receptor at the same site with similar binding patterns. Most of the time, these proteins belong to the same protein family and have a high degree of sequence similarity. However, these proteins can sometimes be scattered among a diverse family of proteins, maintaining sequence conservation for a short conserved sequence or even no apparent sequence conservation. For one thing, the **short conserved sequence** between the disordered protein and the receptor is called **sequence motif**, such as the TRAF6 and its substrates (shown in Figure 1A). For another thing, some **small sub-structures**, called **structural motifs**, are no apparent sequence conservation and scattered among different protein families, such as CRBN and its substrate proteins with different 3D overall structures, as shown in Figure 1B. The development of Motif-Dock was inspired by these PPIs that are difficult to search by sequence alignment methods (e.g. Blast).  

![**Figure 1. Kinds of PPI protein motifs.**](https://user-images.githubusercontent.com/58931275/174751397-d529dfaf-f970-43f2-a0fe-0f3d99c006f7.png)  
**Figure 1. Kinds of PPI protein motifs.**  

Installation
----
Motif-Dock needs to be installed on the Linux system.  

**1. Install the dependent packages.**  
a. Intsall the [Rosetta](https://www.rosettacommons.org/docs/latest/build_documentation/Build-Documentation) packages and set up environment variables.  
```
echo "export rosetta_app=<path to rosetta apps>" >> ~/.bashrc
echo "export rosetta_db=<path to rosetta database>" >> ~/.bashrc
echo "export rosetta_version=<rosetta versions, mpi or static>" >> ~/.bashrc
```  
b. Intsall Julia and related packages.   
```  
wget https://julialang-s3.julialang.org/bin/linux/x64/1.5/julia-1.5.3-linux-x86_64.tar.gz
tar zxvf julia-1.5.3-linux-x86_64.tar.gz  
cd julia-1.5.3/bin
echo "export PATH=${PWD}:\$PATH" >> ~/.bashrc
source ~/.bashrc
julia
] # in Julia REPL
add BioStructures # in Julia REPL
add PDBTools # in Julia REPL
exit()
```
**2. Download Motif-Dock and configure environment variables.**   
```
git clone https://github.com/Wang-Lin-boop/Motif-Dock/
cd Motif-Dock/scripts/
echo " # Scripts of Motif-Dock ,added by Motif-Dock, refer to https://github.com/Wang-Lin-boop/Motif-Dock/
export PATH=${PWD}:\$PATH" >> ~/.bashrc
source ~/.bashrc
```

Database
----
**1. Download the library of prepared human protein structures and disorder sequeneces.**

First, download the 3DMotif-Dock structure library, which may take a bit longer.  
```
wget https://bailab.siais.shanghaitech.edu.cn/service/3DMotif-Dock-pds-Library.zip
unzip 3DMotif-Dock-pds-Library.zip
for pds in `ls 3DMotif-Dock-pds-Library`;do
  echo "${PWD}/3DMotif-Dock-pds-Library/${pds}" >> 3DMotif-Dock-pds-Library.list
done
echo " ## Database of 3DMotif-Dock, added by Motif-Dock, refer to https://github.com/Wang-Lin-boop/Motif-Dock/
export HumanProteinPDS=${PWD}/3DMotif-Dock-pds-Library.list" >> ~/.bashrc
```
Next, switch to the ``Motif-Dock/Database`` and execute the following command.
```
echo " ## Database of 2DMotif-Dock, added by Motif-Dock, refer to https://github.com/Wang-Lin-boop/Motif-Dock/
export HumanDisorderSequence50=${PWD}/Human-Disorder50-protein.fasta 
export HumanDisorderSequence70=${PWD}/Human-Disorder70-protein.fasta" >> ~/.bashrc
```
Finally, three environment variables will be added to your ``~/.bashrc`` file.

**2. Use the 3DMotif-Dock to generate your own structure libraries.**

  First, you need to build your own library of single chain protein structures, which may be obtained from PDB, AlphaFold DB or some conformational ensemble generated by molecular simulation. Next, run the following command to generate the processed structure library.  
```
3DMotif-Dock -i <INPUT_PDB_LIB> -n 40 -1 
```
  The program generates a file named INPUT_PDB_LIB-Index, which will be used in the next calculations as the database index file.  

**3. Use the scripts to generate your own disorder sequences libraries.**

Usage of 2DMotif-Dock
----


Usage of 3DMotif-Dock
----


Citation
----


Acknowledgements
----

